x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/documents_logic.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+export interface DocumentsLogicValues {
+  isDocumentCreationOpen: boolean;
+}
+
+export interface DocumentsLogicActions {
+  closeDocumentCreation(): void;
+  openDocumentCreation(): void;
+}
+
+type DocumnetsLogicType = MakeLogicType<DocumentsLogicValues, DocumentsLogicActions>;
+
+export const DocumentsLogic = kea<DocumnetsLogicType>({
+  path: ['enterprise_search', 'app_search', 'documents_logic'],
+  actions: () => ({
+    openDocumentCreation: true,
+    closeDocumentCreation: true,
+  }),
+  reducers: () => ({
+    isDocumentCreationOpen: [
+      false,
+      {
+        openDocumentCreation: () => true,
+        closeDocumentCreation: () => false,
+      },
+    ],
+  }),
+});
```

Comment:
Oh actually! I probably should have started with this and not a typo sorry, but I'd like to update the Document Creation modal to handle its own open/close state instead of relying on parent views to do it for them. Right now this logic is duplicated inside EngineOverviewLogic and OnboardingLogic as well and it definitely feels like it should just be managed in one place.

That being said I'm about 2 PRs away still from making that happen, so feel free to leave this as-is until then (I'll probably just like pull this file out entirely and move it to a new `document_creation` folder)